python/cudf/cudf/core/column/numerical.py

```diff
@@ -597,7 +597,9 @@ def can_cast_safely(self, to_dtype: DtypeObj) -> bool:
                     # TODO: replace np.inf with cudf scalar when
                     # https://github.com/rapidsai/cudf/pull/6297 merges
                     non_infs = s[
-                        ((s == np.inf) | (s == -np.inf)).logical_not()
+                        ((s == np.inf) | (s == -np.inf))
```

Comment:
unrelated but should probably create a `cudf.inf` singleton device scalar

Reply:
we technically have one:

```
>>> import numpy as np
>>> cudf.Scalar(np.inf)
Scalar(inf, dtype=float64)
```
It just doesn't end up being helpful here because we actually need `inf` and `-inf` so the same scalar isn't reused.